bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/api/SwitchService.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.revogi.internal.api;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.jetbrains.annotations.NotNull;
+import org.openhab.binding.revogi.internal.udp.UdpResponse;
+import org.openhab.binding.revogi.internal.udp.UdpSenderService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link SwitchService} enables the binding to actually switch plugs on and of
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class SwitchService {
+
+    private static final String UDP_DISCOVERY_QUERY = "V3{\"sn\":\"%s\", \"cmd\": 20, \"port\": %d, \"state\": %d}";
+    private static final String VERSION_STRING = "V3";
+    private final Logger logger = LoggerFactory.getLogger(SwitchService.class);
+
+    private final Gson gson = new GsonBuilder().create();
+    private final UdpSenderService udpSenderService;
+
+    public SwitchService(UdpSenderService udpSenderService) {
+        this.udpSenderService = udpSenderService;
+    }
+
+    public CompletableFuture<SwitchResponse> switchPort(String serialNumber, String ipAddress, int port, int state) {
+        if (state < 0 || state > 1) {
+            logger.warn("state value is not valid: {}", state);
+            throw new IllegalArgumentException("state has to be 0 or 1");
```

Comment:
As the unchecked exception is logged by the framework, the additional logging could be removed. Same for below.